src/historywork/DownloadBucketsWork.cpp

```diff
@@ -92,7 +93,9 @@ DownloadBucketsWork::yieldMoreWork()
         std::static_pointer_cast<DownloadBucketsWork>(shared_from_this()));
 
     auto currId = mIndexId++;
+    mIndexMapMutex.lock();
     auto [indexIter, inserted] = mIndexMap.emplace(currId, nullptr);
+    mIndexMapMutex.unlock();
```

Comment:
In general, I think it is a better practice to use `std::lock_guard` rather than `lock` / `unlock` pairs, as the former guarantees that the mutex is unlocked even if there is an exception thrown in between.